gflownet/envs/crystals/miller.py

```diff
@@ -0,0 +1,40 @@
+"""
+Class to represent an environment to sample Miller indices (hkl).
+"""
+from gflownet.envs.grid import Grid
+
+
+class MillerIndices(Grid):
+    """
+    The Miller indices are represented by either 3 parameters (h, k, l) if the
+    structure is cubic, or by 4 parameters (h, k, i, l) if the structure is hexagonal
+    or rhombohedral. Each parameter can take values in the set {-2, -1, 0, 1, 2}.
+    Therefore, we can represent the Miller indices environment by a hyper cube of
+    length 5, with dimensionality 3 or 4 depending on the structure.
+
+    Attributes
+    ----------
+    is_cubic : bool
+        True if the structure is cubic, False if the structure is hexagonal or
+        rhombohedral.
```

Comment:
What if the structure is neither cubic, hexagonal or rhombohedral?
As far as I can tell, those remaining cases would also miller indices specified by 3 parameters like the cubic

Reply:
Yes, right. I guess by cubic I meant non-hexagonal/non-rhombohedral but it is not a good choice for the variable name.